### PR TITLE
Try resolving jobs from service provider before resorting to ActivatorUtilities

### LINKED ISSUE #1157: Can no longer register jobs via DI
See https://github.com/openiddict/openiddict-core/issues/1243#issuecomment-816019916 . Ideally we can support both discovery mechanisms.

### PR DESCRIPTION
**Pull request by lahma**
fixes #1157

### CODE REVIEW

**Comment by kevinchalet** [comment]
Should `(InnerJob as IDisposable)?.Dispose();` be behind a `if` to prevent double-disposal if the service comes from DI? (when it's created using `ActivatorUtilities`, it must be disposed of manually)

**Comment by lahma** [comment]
> Should `(InnerJob as IDisposable)?.Dispose();` be behind a `if` to prevent double-disposal if the service comes from DI? (when it's created using `ActivatorUtilities`, it must be disposed of manually)

Good catch, added check for job's origin and only disposing if it came from ? ActivatorUtilities`.

**Comment by ArturDorochowicz** [comment]
Looks good.

File: src/Quartz.Extensions.DependencyInjection/ServiceCollectionQuartzConfigurator.cs [hunk 1]
```diff
@@ -61,15 +61,10 @@ public void UseMicrosoftDependencyInjectionJobFactory(Action<JobFactoryOptions>?
             UseJobFactory<MicrosoftDependencyInjectionJobFactory>(configure);
         }
 
+        [Obsolete("Jobs are always created with some")]
```
Comment (ArturDorochowicz) [inline src/Quartz.Extensions.DependencyInjection/ServiceCollectionQuartzConfigurator.cs:64]: Did you mean `... with scope`?

Comment (lahma) [reply src/Quartz.Extensions.DependencyInjection/ServiceCollectionQuartzConfigurator.cs]: Thanks, working in parallel with two bug fixes seems to freeze the brain. Pushed fix commit.

**Comment by lahma** [comment]
I think we have the pieces together to patch the DI resolution support.. @kevinchalet and @ArturDorochowicz  are you happy with this so I can merge? Thank you both for helping out and spotting problems.

**Comment by ArturDorochowicz** [comment]
@lahma I think it addresses all concerns. Ship it! :)
Thank you.

**Comment by kevinchalet** [comment]
@lahma looks awesome, thanks for fixing it! 👏🏻

**Pull request merged**